### DISCUSSION TITLE
Denial of Service via Hash Collision Attack

### DISCUSSION THREAD

**Discussion by teb510** [Kong/kong#13306] [Announcements]
## Denial of Service via Hash Collision Attack

Recently, Kong’s security researchers discovered a vulnerability in OpenResty, a third party dependency of Kong Gateway. This vulnerability creates the possibility of Denial of Service attacks against Kong Gateway.

## Vulnerability Description:

Kong's security researchers identified a weakness in the default string hashing function in OpenResty that can be used to generate large numbers of hash collisions by carefully crafting HTTP requests, which can be used in a Denial of Service attack.

A CVE is expected to be issued. The finding is currently evaluated by Kong at CVSS 9.2.

## Kong Products Impacted:

- **Kong Gateway Enterprise**: 1.6.0.0 and subsequent versions.
- **Kong Konnect** using Kong Gateway Enterprise: 2.5.0.1 and subsequent versions.
- **Kong Gateway Community/OSS**: 2.4.0 and above.

## Remediation Steps:

Patches for all supported Kong Gateway versions are available now for both OSS and Enterprise. Additionally, Enterprise customers will have already received communication from Kong regarding this vulnerability. Our recommendation is to apply the latest patch or upgrade Kong Gateway.

For OSS, we have provided patches for 3.7.1 (latest) and also 2.8.5 as we have a large number of community members on this version as well.

## Links to Available Patches:

All available patches can be found at [packages.konghq.com](https://packages.konghq.com). The table below provides direct links to the latest patches for each supported version of Kong Gateway.

### Impacted Kong Gateway Enterprise version:

| Supported Versions | Latest Patch |
|--------------------|--------------|
| 2.8.x.y            | [2.8.4.11](https://cloudsmith.io/~kong/repos/gateway-28/packages/)     |
| 3.4.x.y            | [3.4.3.11](https://cloudsmith.io/~kong/repos/gateway-34/packages/)     |
| 3.5.x.y            | [3.5.0.6](https://cloudsmith.io/~kong/repos/gateway-35/packages/)      |
| 3.6.x.y            | [3.6.1.6](https://cloudsmith.io/~kong/repos/gateway-36/packages/)      |
| 3.7.x.y            | [3.7.1.1](https://cloudsmith.io/~kong/repos/gateway-37/packages/)      |

### Impacted Kong Gateway Community / OSS version:

| Versions Impacted | Latest Patch    |
|--------------|----------------------|
| 2.8.x         | [2.8.5](https://cloudsmith.io/~kong/repos/gateway-28/packages/)  |
| 3.4.x, 3.5.x, 3.6.x, 3.7.x        | [3.7.1](https://cloudsmith.io/~kong/repos/gateway-37/packages/) |

**Comment by jeremyjpj0916** [comment]
Nice to see Kong patch the 2.8.5 OSS version too for folks stuck on older Kong pre 3.x

**Comment by git-hulk** [comment]
@teb510 cool, any plan to patch other minor versions like 3.4 / 3.5 / 3.6?

**Reply by guanlan** [reply]
@git-hulk  We currently do not have plans for further backporting. We highly encourage upgrading to version 3.7.

**Reply by git-hulk** [reply]
@guanlan Got it, thanks for your reply!
